src/vs/editor/contrib/codeAction/browser/codeActionUi.ts

```diff
@@ -31,16 +31,21 @@ export class CodeActionUi extends Disposable {
 		quickFixActionId: string,
 		preferredFixActionId: string,
 		private readonly delegate: {
-			applyCodeAction: (action: CodeActionItem, regtriggerAfterApply: boolean) => Promise<void>;
+			applyCodeAction: (action: CodeActionItem, regtriggerAfterApply: boolean, preview: boolean) => Promise<void>;
 		},
 		@IInstantiationService instantiationService: IInstantiationService,
 	) {
 		super();
 
 		this._codeActionWidget = new Lazy(() => {
 			return this._register(instantiationService.createInstance(CodeActionMenu, this._editor, {
-				onSelectCodeAction: async (action) => {
-					this.delegate.applyCodeAction(action, /* retrigger */ true);
+				onSelectCodeAction: async (action, trigger) => {
+					if (trigger.preview) {
+						this.delegate.applyCodeAction(action, /* retrigger */ true, true);
```

Comment:
Instead of a conditional, you should be able to pass `trigger.preview` as the last argument to `applyCodeAction` to avoid duplicating this code